src/VisualStudio/LiveShare/Impl/Shims/TypeScriptHandlerShims.cs

```diff
@@ -0,0 +1,210 @@
+            return response.Select(obj => (SymbolInformation)obj).ToArray();
+        }
+    }
+
+    [ExportLspRequestHandler(LiveShareConstants.TypeScriptContractName, Methods.TextDocumentFormattingName)]
+    internal class TypeScriptFormatDocumentHandlerShim : FormatDocumentHandler, ILspRequestHandler<DocumentFormattingParams, TextEdit[], Solution>
+    {
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        public TypeScriptFormatDocumentHandlerShim(IThreadingContext threadingContext)
+        {
+            _threadingContext = threadingContext;
+        }
+
+        public Task<TextEdit[]> HandleAsync(DocumentFormattingParams request, RequestContext<Solution> requestContext, CancellationToken cancellationToken)
+            => base.HandleRequestAsync(requestContext.Context, request, requestContext.ClientCapabilities?.ToObject<ClientCapabilities>(), cancellationToken);
+
+        protected override async Task<IList<TextChange>> GetFormattingChangesAsync(IEditorFormattingService formattingService, Document document, TextSpan? textSpan, CancellationToken cancellationToken)
+        {
+            // TypeScript expects to be called on the UI thread to get the formatting options.
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            return await base.GetFormattingChangesAsync(formattingService, document, textSpan, cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    [ExportLspRequestHandler(LiveShareConstants.TypeScriptContractName, Methods.TextDocumentRangeFormattingName)]
+    internal class TypeScriptFormatDocumentRangeHandlerShim : FormatDocumentRangeHandler, ILspRequestHandler<DocumentRangeFormattingParams, TextEdit[], Solution>
+    {
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        public TypeScriptFormatDocumentRangeHandlerShim(IThreadingContext threadingContext)
+        {
+            _threadingContext = threadingContext;
+        }
+
+        public Task<TextEdit[]> HandleAsync(DocumentRangeFormattingParams request, RequestContext<Solution> requestContext, CancellationToken cancellationToken)
+            => base.HandleRequestAsync(requestContext.Context, request, requestContext.ClientCapabilities?.ToObject<ClientCapabilities>(), cancellationToken);
+
+        protected override async Task<IList<TextChange>> GetFormattingChangesAsync(IEditorFormattingService formattingService, Document document, TextSpan? textSpan, CancellationToken cancellationToken)
+        {
+            // TypeScript expects to be called on the UI thread to get the formatting options.
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+            return await base.GetFormattingChangesAsync(formattingService, document, textSpan, cancellationToken).ConfigureAwait(false);
```

Comment:
If this is following JTF rules we don't really need the ConfigureAwait(false) here.

Reply:
There's a warning if I leave it out.  I don't think it should be true though right?

Reply:
Yeah, probably not worth fighting. We could disable the warning but this is all going away.